meta-resin-common/recipes-containers/docker-disk/docker-resin-supervisor-disk/update-resin-supervisor

```diff
@@ -60,11 +60,6 @@ done
 # A temporary file used until next reboot
 UPDATECONF=/tmp/update-supervisor.conf
 
-if [ -z "$API_ENDPOINT" ]; then
-    echo "Environment variable API_ENDPOINT must be set."
-    exit 1
-fi
-
 # If the user api key exists we use it instead of the deviceApiKey as it means we haven't done the key exchange yet
 _device_api_key=${PROVISIONING_API_KEY:-$DEVICE_API_KEY}
```

Comment:
@agherzan this will fail as well for the lack of variables, right? And thus the script doesn't even get to further down.

Reply:
Got it. So how does it fail?

Reply:
```
root@a058750:/tmp# bash update-resin-supervisor
PROVISIONING_API_KEY or DEVICE_API_KEY variables must be set.
```
right as it says in this section of the code that I'm linking to:
```
# If the user api key exists we use it instead of the deviceApiKey as it means we haven't done the key exchange yet
_device_api_key=${PROVISIONING_API_KEY:-$DEVICE_API_KEY}
if [ -z "$_device_api_key" ]; then
    echo "PROVISIONING_API_KEY or DEVICE_API_KEY variables must be set."
    exit 1
fi
```

Reply:
Now I see what you are pointing to. Actually, that is not the case because it is always generated by the system with a systemd service. So basically DEVICE_API_KEY will always be available.

Reply:
@agherzan if that's true, then yes, my test is wrong.

Also, then in the PR
```
 [ -n "$API_ENDPOINT" ] && [ -n "$DEVICE_ID" ] && [ -n "$_device_api_key" ]
```
should be just
```
 [ -n "$API_ENDPOINT" ] && [ -n "$DEVICE_ID" ] 
```
right?

How about the `DEVICE_ID`, is that always available too?

Reply:
DEVICE_ID is generated by the registration. Which can happen either through the supervisor or thought the resin-device-register (in the case of flasher). So if you don't have it, it means that the device didn't register yet.

Now about _device_api_key. I included it in that script because the curl command uses it later. So if it is empty it makes no sense to try to curl. Similar to the other ones.

On my last comment about the existence of DEVICE_API_KEY, it seems I was wrong. Actually, I wasn't wrong but it was not what actually happens. The supervisor service has a dependency on resin-device-api-key.service which generates the device API key. This service (resin-supervisor) triggers the update-resin-supervisor when there is no image available (supervisor was not preloaded) so basically update-supervisor service will have it available.

See:
https://github.com/resin-os/meta-resin/blob/master/meta-resin-common/recipes-containers/docker-disk/docker-resin-supervisor-disk/resin-supervisor.service
https://github.com/resin-os/meta-resin/blob/master/meta-resin-common/recipes-containers/docker-disk/docker-resin-supervisor-disk/update-resin-supervisor.service
https://github.com/resin-os/meta-resin/blob/master/meta-resin-common/recipes-support/resin-unique-key/resin-unique-key/resin-device-api-key.service

Reply:
@agherzan but then I'm confused, these two statements are in conflict for me, from your earlier replies:

* "So basically DEVICE_API_KEY will always be available.", so that means `_device_api_key` can never be empty, thus bailing on non-existent keys is good earlier in the script 
* and above here you say "So if it (`_device_api_key`) is empty it makes no sense to try to curl." but didn't you say it can never be empty, and if it was, the earlier part of the script would have `exit 1` already?

Reply:
@imrehg It's just an extra check. Basically, if anything needed by the curl query is missing, avoid running the query. It happens that indeed, the device will always be there, but that might change.

Reply:
@agherzan I don't like this style, as the code with these two contradicting sections is incomprehensible without any further explanation, and that extra check is completely redundant to the extent that it is more confusing than helpful.  I understand your "extra check" and purpose explanation, and on its own it's very good. The two parts together that are weird.

I guess it works, but to me confusion like this is the sign of the intentions in the code are not completely clear, which makes it more difficult to check whether things are working as intended in the future.